src/adam7.rs

```diff
@@ -78,16 +114,21 @@ impl Adam7Iterator {
     fn init_pass(&mut self) {
         let w = f64::from(self.width);
         let h = f64::from(self.height);
-        let (line_width, lines) = match self.current_pass {
-            1 => (w / 8.0, h / 8.0),
-            2 => ((w - 4.0) / 8.0, h / 8.0),
-            3 => (w / 4.0, (h - 4.0) / 8.0),
-            4 => ((w - 2.0) / 4.0, h / 4.0),
-            5 => (w / 2.0, (h - 2.0) / 4.0),
-            6 => ((w - 1.0) / 2.0, h / 2.0),
-            7 => (w, (h - 1.0) / 2.0),
-            _ => unreachable!(),
+        let (line_mul, line_off, samp_mul, samp_off) = {
```

Comment:
I think this whole function can be simplified a bunch. If I'm not missing something, should be a matter of...

```rust
let pass = PASS_CONSTANTS[self.current_pass as usize - 1];

self.line_width = (self.width - pass.samp_off).div_ceil(pass.samp_mul);
self.lines = (self.width - pass.line_off).div_ceil(pass.line_mul);
self.line = 0;
```

(And making the constants `u32` rather than `usize`)